### ISSUE TITLE
Harmonising method names 

### ISSUE THREAD

**Issue by opeltre** [geomstats/geomstats#412]
It would be nice to harmonise method names! Following an overall common scheme and finding short transparent names would make object's interfaces be easier to learn and remember.

For instance, we could start using "to" and "from" more systematically (actual names in comment):
```python
manifold.is_point                     # belongs
manifold.to_point                    #  regularize?

manifold.is_tangent                 # is_tangent_vector 
manifold.to_tangent               # projection_to_tangent_space

matrices.is_symmetric             
matrices.to_symmetric            # make_symmetric
```
Any suggestion welcome below :)

**Comment by nguigs** [comment]
Definitely agree that is_tangent/to_tangent are necessary and should be added to as many manifolds as possible. Also agree that regularize is not informative, so belongs can be changed to is_point to match the others.

**Comment by stefanheyder** [comment]
I also agree! `is_X` and `to_X` are good naming conventions 

* you can see the return type from the method name
* `regularize` is ambiguous 
* uniform naming is a good thing to have!

**Comment by opeltre** [comment]
For SO, SE, SPD: 
+ `from_vector`  ? 
+ `to_vector` ? 
Converting symmetric/skew-symmetric matrices to 1D-array using a basis

**Comment by nguigs** [comment]
`from_1d` ? `to_1d`?

**Comment by ninamiolane** [comment]
Agreed that uniformizing methods names is great, and `from_`, `to_` naming is explicit! 

Some remarks:

1. What about the methods that can be in several manifolds: `matrices.is_symmetric, matrices.to_symmetric` can also be in symmetric_matrices: `sym_matrices.is_point`, `sym_matrices.to_point`?
    - I think we should keep both, but factorize the code in auxiliary functions, in order to avoid copy-pasting and changing the code at 2 locations when we want to change its logic.

2. The methods `.to_vector`, `.from_vector` are great to change `point_type`, i.e. moving back and forth between `vector` and `matrix` representations. Can we adapt this naming scheme to `coords_types`, i.e. the different coordinate systems with which we can represent a point? 

For example, a point in hyperbolic space can be represented in `explicit`, `implicit`, or `ball` coordinates, which are all `point_type='vector'` representations. Writing `hyperbolic.to_ball`, is not explicit enough, and we do not know from which coords_type we are converting. 

Several solutions:
    a. Add `from` in `hyperbolic.to_ball(x, from='extrinsic')`, and `to` in `hyperbolic.from_ball(x, to='intrinsic')`
    b.  **Current solution in the codebase:** Create a method `change_coordinates_system(x, from='extrinsic', to='intrinsic')`. @tgeral68 
    c. Enforce that each manifold **object** has a unique `default_point_type`/`default_coords_type`: 
        - `hyperbolic.to_ball` and `hyperbolic.from_ball` functions convert from and to the default_point_type/coords_type chosen when instantiating `hyperbolic`.
    d. Enforce that each manifold **class** has a unique `default_point_type`/`default_coords_type`:
        - Same logic.
--> I would prefer option 2c), as a compromise between organizing the code but not creating too many classes, or option 2d).
--> All conversion methods could be private functions that go into separate python files: e.g. `hyperbolic_coords.py`, `special_orthogonal_coords.py`, etc.

3. What do we want `space.to_point` represent?
    a. the "canonical" projection to space? 
        - E.g. the orthogonal projection for its "canonical"/default metric, that we assign to it?
    b. the choice of a representative in an equivalence class?
        - E.g. the current `regularize`, for SO(3), is taking the modulo Pi in the rotation vector's norm. This case will arise with quotient manifolds soon as well.
I would prefer option 3a), and give option 3b) a different name.
**Remark**: If we use 2c) or 2d) options, we do not need to add `point_type` and `coord_type` parameters to `space.to_point` 👍 .

**Comment by nguigs** [comment]
1. I think the idea of the Matrices class was to gather all such useful methods (as class methods) that could be used anywhere else in the code base. Matrix subspaces should then inherit these. This means the method names in Matrices are explicit (e.g. make_symmetric), and other spaces that we want as represent faithful to geometry can call these methods with specific name (e.g. project in Symmetric spaces)
2. a or 2.b seem more explicit, which is fine I think, as changing coordinates is common in a manifold, whereas to/from_ball is not very intuitive, and you have to look for the specific method every time. and 2.c has a potentiel overhead if you always convert to a default_point_type
I think 3.b corresponds more to a lift (or a section) than to a projection

**Comment by ninamiolane** [comment]
1. Agreed! Putting in Matrices seems great, l can work on this.
2. OK, let's keep the current implementation for now.
3. Agreed!

**Comment by ninamiolane** [comment]
I started to address this issue in PR #619. 

Even if `belongs` and `projection` are less "harmonized" with the other methods' names, I think that they are more explicit and intuitive than `is_point`, `to_point`. I haven't changed them for now. 

Do you think a user would prefer `is_point` and `to_point`?

**Comment by kellertuer** [comment]
For m `to_point` would not determine, _how_ one obtains a point from the input, so I would say `projection` is better (or even `project`? that might be confused with its noun however).

`belongs` sounds nice, I don't know whether it sounds better than `is_point`. But together with `is_tangent`, `is_point` sounds good.

**Comment by ninamiolane** [comment]
Agreed on the difference between `to_point` and `project`: following this logic, we should also find another name for `to_tangent`, currently used in the codebase!

I'm just worried that changing `belongs` to `is_point` might be confusing when `points` are in fact `matrices` or `curves`, for example:
```discrete_curves_space.belongs(my_curve)```
reads better than:
```discrete_curves_space.is_point(my_curve)```.

**Comment by kellertuer** [comment]
In our system (disclaimer: different language https://juliamanifolds.github.io/Manifolds.jl/latest/interface.html#ManifoldsBase.project-Tuple{Manifold,Any,Any} ) we have a `project` for both points and tangents, but they can be distinguished by number of arguments – `project(M,p)` for points, `project(M,p,X)` for tangents.
But `to_tangent` also does not specify what (magic) is happening to get a tangent.
In Python with an object my (biased) idea would be `M.project(p)` or `M.project(p,X)`.  Might that work in Python? Otherwise one could use `project_point` and `project_tangent`, which I found too long.

For the second, oh that's a tricky one, since abstractly you have manifolds and points, but in the concrete case, yes, the point is a matrix or a curve or even a shape. Then these terms might clash. Then, in the object setting, you're right, the belongs seems to be more general in readability. I stumbled, since it seems in wrong order (M.belongs(p) reads the wrong way around maybe?); one could do `M.validate`, again even for both points and tangents.
